File: platforms/kotlin/examples/scene/light/set_direction.kt
import org.fragmentcolor.*

val sun = Light.directional(listOf(0.0f, -1.0f, 0.0f), listOf(1.0f, 1.0f, 1.0f))
sun.setDirection(listOf(0.3f, -0.8f, -0.5f))

// Point lights have no direction — the call errors.
val lamp = Light.point(listOf(0.0f, 2.0f, 0.0f), listOf(1.0f, 1.0f, 1.0f))
val result = lamp.setDirection(listOf(0.0f, -1.0f, 0.0f))

File: platforms/python/examples/scene/light/set_direction.py
from fragmentcolor import Light

sun = Light.directional([0.0, -1.0, 0.0], [1.0, 1.0, 1.0])
sun.set_direction([0.3, -0.8, -0.5])

# Point lights have no direction — the call errors.
lamp = Light.point([0.0, 2.0, 0.0], [1.0, 1.0, 1.0])
result = lamp.set_direction([0.0, -1.0, 0.0])